src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs

```diff
@@ -506,20 +507,24 @@ private bool LaunchHost(TestProcessStartInfo testHostStartInfo, CancellationToke
             return false;
         }
 
-        SetProcessPriority(_testHostProcess, _environmentVariableHelper);
+        AdjustProcessPriorityBasedOnSettings(_testHostProcess, testHostStartInfo.EnvironmentVariables);
         OnHostLaunched(new HostProviderEventArgs("Test Runtime launched", 0, _testHostProcess.Id));
 
         return true;
     }
 
-    internal static void SetProcessPriority(Process testHostProcess, IEnvironmentVariableHelper environmentVariableHelper)
+    internal static void AdjustProcessPriorityBasedOnSettings(Process testHostProcess, IDictionary<string, string?>? testHostEnvironmentVariables)
     {
         ProcessPriorityClass testHostPriority = ProcessPriorityClass.BelowNormal;
         try
         {
-            testHostPriority = environmentVariableHelper.GetEnvironmentVariableAsEnum("VSTEST_HOST_INTERNAL_PRIORITY", testHostPriority);
-            testHostProcess.PriorityClass = testHostPriority;
-            EqtTrace.Verbose("Setting test host process priority to {0}", testHostProcess.PriorityClass);
+            if (testHostEnvironmentVariables is not null
```

Comment:
@Evangelink @nohwnd I am not familiar with the code base - but is DefaultTestHostManager also used when parallel is not in play?

Just want to make sure that it is possible to create low prio test hosts -
a) regardless of whether parallel is turned on and
b) regardless of whether the operation requested is execution versus discovery

Also if the answer to b) above is yes, then it may be better to rename this to something more general like VSTEST_BACKGROUND_OPERATION or VSTEST_BACKGROUND_PRIORITY :)

Reply:
Yes that code is used even in "sequential". This is used to decide on the level of parallelism (including sequential) to use.